main.tf

```diff
@@ -25,10 +34,13 @@ resource "google_sql_database_instance" "main" {
       private_network = var.private_network
     }
 
-    backup_configuration {
-      enabled            = var.backup_configuration_enabled
-      binary_log_enabled = var.backup_configuration_binary_log_enabled
-      start_time         = "02:39"
+    dynamic "backup_configuration" {
+      for_each = var.backup_configuration_enabled ? [1] : []
```

Comment:
any specific reason why we need for_each? Not sure if you can have multiple backup configuration for the same instance.

Reply:
the reason is to create backups only when we create the primary instance because we don't need it for the replicas